packages/cli/generators/model/index.js

```diff
@@ -0,0 +1,273 @@
+          message: 'Is ID field?',
+          type: 'confirm',
+          default: false,
+          when: function(answers) {
+            return (
+              !this.idFieldSet &&
+              !['array', 'object', 'buffer'].includes(answers.type)
+            );
+          }.bind(this),
+        },
+        {
+          name: 'required',
+          message: 'Required?:',
+          type: 'confirm',
+          default: false,
+        },
+        {
+          name: 'default',
+          message: `Default value ${chalk.yellow('[leave blank for none]')}:`,
+          when: function(answers) {
+            return (
+              ![null, 'buffer', 'any'].includes(answers.type) &&
+              this.typeChoices.includes(answers.type)
+            );
+          }.bind(this),
+        },
+      ];
+
+      return this.prompt(prompts).then(props => {
+        if (props.default === '') {
+          delete props.default;
+        }
+
+        Object.assign(this.artifactInfo.properties[this.propName], props);
+        if (props.id) {
+          this.idFieldSet = true;
+        }
+
+        this.log(
+          `Let's add another property to ${chalk.yellow(
+            this.artifactInfo.className,
+          )}`,
+        );
+        return this.promptPropertyName();
+      });
+    }
+  }
+
+  scaffold() {
+    if (this.shouldExit()) return false;
+
+    // Data for templates
+    this.artifactInfo.baseClassName = `${this.artifactInfo.className}Base`;
+    this.artifactInfo.fileName = utils.kebabCase(this.artifactInfo.name);
+    this.artifactInfo.jsonFileName = `${this.artifactInfo.fileName}.model.json`;
+    // prettier-ignore
+    this.artifactInfo.baseOutFile = `${this.artifactInfo.fileName}.base.model.ts`;
+    this.artifactInfo.outFile = `${this.artifactInfo.fileName}.model.ts`;
+
+    // Resolved Output Paths
+    const jsonPath = this.destinationPath(
+      this.artifactInfo.outDir,
+      this.artifactInfo.jsonFileName,
+    );
+    const tsPath = this.destinationPath(
+      this.artifactInfo.outDir,
+      this.artifactInfo.outFile,
+    );
+    const baseTsPath = this.destinationPath(
+      this.artifactInfo.outDir,
+      '_generated',
+      this.artifactInfo.baseOutFile,
+    );
+
+    this.artifactInfo.updateIndexes = [
+      {dir: this.artifactInfo.outDir, file: this.artifactInfo.outFile},
+      {
+        dir: path.resolve(this.artifactInfo.outDir, '_generated'),
+        file: this.artifactInfo.baseOutFile,
+      },
+    ];
+
+    const baseModelTemplatePath = this.templatePath('base.model.ts.ejs');
+    const modelTemplatePath = this.templatePath('model.ts.ejs');
+
+    // Model JSON
+    const model = _.cloneDeep({
+      name: this.artifactInfo.className,
+      extends: this.artifactInfo.base,
+      properties: this.artifactInfo.properties,
+    });
+
+    // Set up types for Templating
+    const TS_TYPES = ['string', 'number', 'object', 'boolean', 'any'];
+    const NON_TS_TYPES = ['geopoint', 'date'];
```

Comment:
I thought date type would be where we'd use JavaScript's own `Date` like Buffer.
I think we should have something like `const COMPLEX_TS_TYPES = ['buffer', 'date']` and then have a matching series of if statements to go with it